src/libraries/Microsoft.Extensions.Http/src/LifetimeTrackingHttpMessageHandler.cs

```diff
@@ -18,6 +18,11 @@ public LifetimeTrackingHttpMessageHandler(HttpMessageHandler innerHandler)
         protected override void Dispose(bool disposing)
         {
             // The lifetime of this is tracked separately by ActiveHandlerTrackingEntry
+            // During explicit disposal, we need to dispose the inner handler
+            if (disposing)
+            {
+                base.Dispose(disposing);
+            }
```

Comment:
Please revert that.
I see that you attempted to fix the issue of the handlers not being disposed, but the thing is: we don't (shouldn't) dispose LifetimeTrackingHttpMessageHandler directly; when disposing an entry, we are (should be) disposing LifetimeTrackingHttpMessageHandler's inner handler